test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/MultipleDisposalsTriageControllerSpec.scala

```diff
@@ -1775,42 +1874,58 @@ class MultipleDisposalsTriageControllerSpec
         "the user has not started a new draft return and" when {
 
           "the journey is incomplete" in {
-            List(trustDisplay, agentDisplay, individualDisplay).foreach { displayType: TrustAgentOrIndividualDisplay =>
-              withClue(
-                s"For user type ${displayType.userType} with ${displayType.name.fold[String](_ => "Trust Name", _ => "Individual Name")} "
-              ) {
-                test(
-                  sessionDataWithStartingNewDraftReturn(
-                    incompleteAnswers,
-                    displayType.name,
-                    displayType.userType === UserType.Agent
-                  )._1.copy(userType = Some(displayType.userType)),
-                  triage.routes.MultipleDisposalsTriageController.wereYouAUKResident(),
-                  "button.continue",
-                  expectReturnToSummaryLink = false,
-                  displayType
-                )
-              }
+            List(trustDisplay, agentDisplay, individualDisplay, capacitorDisplay, personalRepDisplay).foreach {
+              displayType: UserTypeDisplay =>
+                withClue(
+                  s"For user type ${displayType.getSubKey}"
+                ) {
+                  test(
+                    sessionDataWithStartingNewDraftReturn(
+                      incompleteAnswers.copy(
+                        individualUserType =
+                          if (displayType.representativeType.exists(_.isLeft)) Some(PersonalRepresentative)
+                          else if (displayType.representativeType.exists(_.isRight)) Some(Capacitor)
+                          else Some(Self),
```

Comment:
You are using this logic at couple of places in this Spec. I feel its good to move that logic into a function.

Reply:
ok can do that